### DISCUSSION TITLE
How to go about filtering on fields that require multiple joins

### DISCUSSION THREAD

**Discussion by bryant-roach7** [arthurio/fastapi-filter#245] [Q&A]
I need to specify a filter that is from UserFilter, for example, and have it applied to all my tables even ones that dont directly connect to the User table. The issue is that in order to apply that filter there need to be multiple joins that take place to link the tables I need to get to the filed I want to filter. What would be the best way to go about that? Does this handle joining tables assuming sqlalchemy knows the relationships?

**Comment by arthurio** [comment]
You have some examples on how to filter related fields (the `AddressFilter` on `UserFilter` in the examples). You want to make sure you are joining all the required tables in your main query that you apply the filter to. For your filters, I would recommend setting a relationship `user` field on all your tables so that you only nest one level.
